src/engine/ui.cpp

```diff
@@ -313,10 +353,13 @@ struct gui : guient
             mergelist = curlist;
             mergedepth = curdepth;
         }
+        if(!curchildpath.empty()) curchildpath.last()++;
+        curchildpath.emplace_back(0);
     }
 
     int poplist()
     {
+        if(curchildpath.size()) curchildpath.pop_back();
```

Comment:
please write `if (int(curchildpath.size()) != 0)` instead of `if (curchildpath.size())`